servant-client-core/src/Servant/Client/Core/Internal/RunClient.hs

```diff
@@ -48,3 +51,12 @@ decodedAs response contentType = do
     Right val -> return val
   where
     accept = toList $ contentTypes contentType
+
+instance ClientF ~ f => RunClient (Free f) where
+    runRequest req  = liftF (RunRequest req id)
+    streamingRequest req = liftF (StreamingRequest req id)
+    throwServantError = liftF . Throw
+    catchServantError x h = go x where
```

Comment:
this is not particularly correct, as `RunRequest` could throw things catchable by `catchServantError` too.